src/lib.rs

```diff
@@ -127,8 +133,8 @@ pub fn run(args: Args) {
             let path =
                 format_relative_path(project_file_dir_entry.path(), args.is_using_default_paths());
 
-            let matcher = TreeSitterMatcher::new(
-                &query,
+            let query_context = QueryContext::new(
```

Comment:
A `QueryContext` is basically just bundling up the same things that were passed to a `TreeSitterMatcher`

I got rid of `Matcher`'s entirely